player/loadfile.c

```diff
@@ -1521,6 +1523,41 @@ static void load_external_opts(struct MPContext *mpctx)
     mp_waiter_wait(&wait);
 }
 
+static void append_to_watch_history(struct MPContext *mpctx)
+{
+    if (!mpctx->opts->save_watch_history)
+        return;
+
+    void *ctx = talloc_new(NULL);
+    char *history_path = mp_get_user_path(ctx, mpctx->global,
```

Comment:
This can be null when `no-config` is used.

Reply:
This is `watch_history.jsonl` with `--no-config --save-watch-history`

Reply:
So what happens it saves history in cwd?

Reply:
Yep.